### ISSUE TITLE
Port proverb exercise from JavaScript track

### ISSUE THREAD

**Issue by rchavarria** [exercism/ecmascript#331]
As part of the job to make the track compatible with *Nextercism*, we identified there are missing exercises in this track that are present in the [JavaScript](https://github.com/exercism/javascript) track. One of them is [proverb](https://github.com/exercism/javascript/tree/master/exercises/proverb).

The task would be to add the `proverb` exercise to this track, copying files from the JavaScript track and adapting the code to the ECMAScript 6 standard.

Basic steps would include:

1. Copy all files from `<javascript repo>/exercises/proverb` (https://github.com/exercism/javascript) to `<ecmascript repo>/exercises/proverb` (https://github.com/exercism/ecmascript)
2. Copy the main `package.json` file (it's in the root folder of the `ecmascript` repo https://github.com/exercism/ecmascript/blob/master/package.json) to the `<ecmascript repo>/exercises/proverb` folder
3. Adapt the code in `proverb.spec.js` file to use ECMAScript 6 features (`import` modules, arrow functions, `const` and/or `let`, any other feature you think will make the code better). Additionally, the JavaScript track uses Jasmine where the ECMAScript track uses Jest.  So, `it` and `xit` must be *translated* to `test` and `xtest`. Feel free to adapt `example.js` as well, just for fun (as long as tests pass)
4. In the `README.md` of the exercise, replace sections from *Setup* and below (Setup, Making the test suite pass, Source and Submitting incomplete solutions) with similar sections from any ECMAScript exercise, for example [`accumulate` README](https://github.com/exercism/ecmascript/blob/master/exercises/accumulate/README.md). They'll include Setup, Requirements, Making the test suite pass, Source and Submitting incomplete solutions.
5. Copy some exercise information from [JavaScript `config.json` file](https://github.com/exercism/javascript/blob/master/config.json) to the [ECMAScript `config.json` file](https://github.com/exercism/ecmascript/blob/master/config.json). This information is an object in the `exercises` array property with a `slug` property equals to `proverb`. Copy that object and place it in the same (or most similar) position in the ECMAScript `config.json` file. Both `config.json` files are in the root folder of the repo.

@exercism/ecmascripts - Do you miss something here?

**Comment by skabbass1** [comment]
@rchavarria @matthewmorgan 
There does not seem to be a [canonical test file](https://github.com/exercism/problem-specifications/tree/master/exercises/proverb) for this exercise set.  Is there a reason for that?   Would it suffice to just port over the tests currently in  the javascript version of the exercise?

**Comment by matthewmorgan** [comment]
@skabbass1 if's there's no canonical file, porting the javascript tests would make sense.

I think the last test ("test the proverb is the same each time") should be implemented differently though.  It _happens_ that in the example implementation in javascript, the `toString()` method rebuilds the entire output from the inputs each time it's called.

My feeling is that it would be better to make two different `Proverb` objects, and compare their output.

@rchavarria ?

**Comment by skabbass1** [comment]
@matthewmorgan  Agree that the current solution is a bit hard to follow. I will refactor and then we can review if it makes sense.

**Comment by matthewmorgan** [comment]
@skabbass1 great.  I think with the ES6 `() =>`, you would easily be able to eliminate the `bind()` usage.

Improvements are always good!

**Comment by rchavarria** [comment]
Yep. I usually find `bind` and `apply` usages a bit hard to read and understand. If there is a simpler solution, I favor the simple one.

Great work @skabbass1!

**Comment by apapirovski** [comment]
This should probably be closed so someone doesn't accidentally work on it. :)

**Comment by joelwallis** [comment]
Closed by #340.
